qa/tasks/cephfs/test_strays.py

```diff
@@ -620,6 +619,42 @@ def test_migration_on_shutdown(self):
         # See that the stray counter on rank 0 has incremented
         self.assertEqual(self.get_mdc_stat("strays_created", rank_0_id), 1)
 
+    def test_migrate_unlinked_dir(self):
+        """
+        Reproduce https://tracker.ceph.com/issues/53597
+        """
+        rank_0_id, rank_1_id = self._setup_two_ranks()
+
+        self.mount_a.run_shell(["mkdir", "pin"])
+
+        self._force_migrate("pin")
```

Comment:
Test fails with:

```
2022-02-05T21:48:59.852 INFO:tasks.cephfs_test_runner:test_migrate_unlinked_dir (tasks.cephfs.test_strays.TestStrays) ... ERROR
2022-02-05T21:48:59.853 INFO:tasks.cephfs_test_runner:
2022-02-05T21:48:59.853 INFO:tasks.cephfs_test_runner:======================================================================
2022-02-05T21:48:59.854 INFO:tasks.cephfs_test_runner:ERROR: test_migrate_unlinked_dir (tasks.cephfs.test_strays.TestStrays)
2022-02-05T21:48:59.854 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2022-02-05T21:48:59.855 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2022-02-05T21:48:59.855 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_09fc2e9e470435f1e6dbf462a51853391f4a4cc0/qa/tasks/cephfs/cephfs_test_case.py", line 365, in _wait_subtrees
2022-02-05T21:48:59.855 INFO:tasks.cephfs_test_runner:    while proceed():
2022-02-05T21:48:59.856 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_3094160cc590b786a43d55faaf7c99d5de71ce56/teuthology/contextutil.py", line 133, in __call__
2022-02-05T21:48:59.856 INFO:tasks.cephfs_test_runner:    raise MaxWhileTries(error_msg)
2022-02-05T21:48:59.856 INFO:tasks.cephfs_test_runner:teuthology.exceptions.MaxWhileTries: reached maximum tries (15) after waiting for 30 seconds
2022-02-05T21:48:59.856 INFO:tasks.cephfs_test_runner:
2022-02-05T21:48:59.857 INFO:tasks.cephfs_test_runner:The above exception was the direct cause of the following exception:
2022-02-05T21:48:59.857 INFO:tasks.cephfs_test_runner:
2022-02-05T21:48:59.857 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2022-02-05T21:48:59.858 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_09fc2e9e470435f1e6dbf462a51853391f4a4cc0/qa/tasks/cephfs/test_strays.py", line 710, in test_migrate_unlinked_dir
2022-02-05T21:48:59.858 INFO:tasks.cephfs_test_runner:    self._force_migrate("pin")
2022-02-05T21:48:59.858 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_09fc2e9e470435f1e6dbf462a51853391f4a4cc0/qa/tasks/cephfs/test_strays.py", line 609, in _force_migrate
2022-02-05T21:48:59.859 INFO:tasks.cephfs_test_runner:    self._wait_subtrees([(rpath, rank)], rank=rank, path=rpath)
2022-02-05T21:48:59.859 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_09fc2e9e470435f1e6dbf462a51853391f4a4cc0/qa/tasks/cephfs/cephfs_test_case.py", line 378, in _wait_subtrees
2022-02-05T21:48:59.859 INFO:tasks.cephfs_test_runner:    raise RuntimeError("rank {0} failed to reach desired subtree state".format(rank)) from e
2022-02-05T21:48:59.859 INFO:tasks.cephfs_test_runner:RuntimeError: rank 1 failed to reach desired subtree state
```

A directory gets migrated to the pinned mds rank if its not empty.

Reply:
So I just create an empty file in the pin directory. Hopes it works.